src/gantry_calibration/calibrate.py

```diff
@@ -0,0 +1,227 @@
+"""
+Gantry Calibration Algorithm Demo.
+
+Usage:
+    calibrate.py basic
+    calibrate.py bilinear [<points>]
+    calibrate.py biquadratic [<points>]
+    calibrate.py bicubic [<points>]
+    calibrate.py poly <x_order> <y_order> <x_points> <y_points>
+    calibrate.py -h | --help
+
+Options:
+    -h --help           Show help screen.
+
+For more information see README.md
+"""
+
+import sys
+from typing import Callable
+from collections import namedtuple
+import numpy as np
+import matplotlib.pyplot as plt  # type: ignore
+from docopt import docopt
+from calibration import Point, PointPair, Calibration
+
+Travel = namedtuple("Travel", ["min", "max"])  # Range of travel of each axis
+
+TRAVEL_MIN = 0.0
+TRAVEL_MAX = 1.0
+TRAVEL_RANGE = TRAVEL_MAX - TRAVEL_MIN
+ERROR_FRACTION = 0.05
+
+
+def main() -> None:
+    """Dispatch command for various interpolation methods."""
+    print("*** Gantry Calibration Algorithm ***")
+
+    arguments = sys.argv[1:]
+    args = docopt(__doc__, argv=arguments)
+
+    if args["basic"]:
+        basic_calibration()
+    if args["bilinear"]:
+        if args["<points>"]:
+            points = int(args["<points>"])
+        else:
+            points = 2
+        polynomial_interpolation(1, 1, points, points)
+    elif args["biquadratic"]:
+        if args["<points>"]:
+            points = int(args["<points>"])
+        else:
+            points = 3
+        polynomial_interpolation(2, 2, points, points)
+    elif args["bicubic"]:
+        if args["<points>"]:
+            points = int(args["<points>"])
+        else:
+            points = 4
+        polynomial_interpolation(3, 3, points, points)
+    elif args["poly"]:
+        polynomial_interpolation(
+            int(args["<x_order>"]),
+            int(args["<y_order>"]),
+            int(args["<x_points>"]),
+            int(args["<y_points>"]),
+        )
+
+
+def basic_calibration() -> None:
```

Comment:
This function is essentially just a copy/paste of the polynomial_interpolation() below. If you really want a separate function for the basic calibration, just call polynomial_interpolation() inside it with pre-defined values. I think it would be more intuitive to delete this function entirely and just call polynomial_interpolation() for all cases in the main function.

Reply:
Deleted function and just call polynomial_interpolation().